security-privacy.html

```diff
@@ -207,90 +211,161 @@
+          for the desired access type (read-write, or read-only),
+          and the URL scopes that are explicitly allowed to access the data,
+          for instance
+          <code>{ "scope": "https://mydomain.com", "write": "true"}</code>.
+          Scopes not matching this are not permitted to access the data.
+          If the list is empty, then the platform default policy applies, which
+          SHOULD permit reading for everyone, but SHOULD prevent writing, except
+          to sub-scopes of the URL scope of the page which has created the data.
+        </li>
+      </ul>
+      <!--
+      The full JSON schema is the following:
+      <pre title = "Web NFC JSON schema" class="example">
+        {
+          "$schema": "http://w3.org/web-nfc#",
+          "id": "http://w3.org/web-nfc",
+          "type": "object",
+          "properties": {
+            "id": {
+              "id": "http://w3.org/web-nfc/id",
+              "type": "string"
+            },
+            "scope": {
+              "id": "http://w3.org/web-nfc/scope",
+              "type": "string"
+            },
+            "messageType": {
+              "id": "http://w3.org/web-nfc/messageType",
+              "type": "string"
+            },
+            "message": {
+              "id": "http://w3.org/web-nfc/message",
+              "type": "string"
+            },
+            "access": {
+              "id": "http://w3.org/web-nfc/access",
+              "type": "array",
+              "items": [
+                {
+                  "id": "http://w3.org/web-nfc/access/0",
+                  "type": "object",
+                  "properties": {
+                    "scope": {
+                      "id": "http://w3.org/web-nfc/access/0/scope",
+                      "type": "string"
+                    },
+                    "access": {
+                      "id": "http://w3.org/web-nfc/access/0/access",
+                      "type": "string"
+                    }
+                  }
+                },
+                {
+                  "id": "http://w3.org/web-nfc/access/1",
+                  "type": "object",
+                  "properties": {
+                    "scope": {
+                      "id": "http://w3.org/web-nfc/access/1/scope",
+                      "type": "string"
+                    },
+                    "access": {
+                      "id": "http://w3.org/web-nfc/access/1/access",
+                      "type": "string"
+                    }
+                  }
+                }
+              ]
+            }
+          },
+          "required": [
+            "id",
+            "scope",
+            "messageType",
+            "message",
+          ]
+        }
+      </pre>
+    </p>
+    -->
+    <p class="note">
+      The encoded JSON format for Web NFC data and metadata is useful for
+      abstracting away NFC Forum types, and for supporting implementations where
```

Comment:
Yes for the first half. No for the second: when the platform supports only one record per message, we need to encode all data there, since we cannot use the first alternative. But then it's not any more a generic tag, i.e. we cannot use records of type "text", "URL", etc, as defined by NFC Forum, instead we need to encode all data into a Web NFC specific record, used as a container.